litestar/middleware/csrf.py

```diff
@@ -63,47 +58,27 @@ def generate_csrf_token(secret: str) -> str:
     return token + token_hash
 
 
-class CSRFMiddleware(MiddlewareProtocol):
+class CSRFMiddleware(ASGIMiddleware):
     """CSRF Middleware class.
 
     This Middleware protects against attacks by setting a CSRF cookie with a token and verifying it in request headers.
     """
 
-    scopes: Scopes = {ScopeType.HTTP}
+    scopes = (ScopeType.HTTP,)
 
-    def __init__(self, app: ASGIApp, config: CSRFConfig) -> None:
+    def __init__(self, config: CSRFConfig) -> None:
         """Initialize ``CSRFMiddleware``.
 
         Args:
-            app: The ``next`` ASGI app to call.
             config: The CSRFConfig instance.
         """
-        self.app = app
         self.config = config
-        self.exclude = build_exclude_path_pattern(exclude=config.exclude, middleware_cls=type(self))
+        self.exclude_opt_key = config.exclude_from_csrf_key
+        self.exclude_path_pattern = config.exclude
 
-    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
-        """ASGI callable.
-
-        Args:
-            scope: The ASGI connection scope.
-            receive: The ASGI receive function.
-            send: The ASGI send function.
-
-        Returns:
-            None
-        """
+    async def handle(self, scope: Scope, receive: Receive, send: Send, next_app: ASGIApp) -> None:
         if scope["type"] != ScopeType.HTTP:
-            await self.app(scope, receive, send)
-            return
-
-        if should_bypass_middleware(
-            scope=scope,
-            scopes=self.scopes,
-            exclude_opt_key=self.config.exclude_from_csrf_key,
-            exclude_path_pattern=self.exclude,
-        ):
-            await self.app(scope, receive, send)
+            await next_app(scope, receive, send)
```

Comment:
Coverage is right; This line will never run as we already define `http` scope in `scopes`. It can just be removed